Test DNSResolver Fails on OSX

<!--- Provide a general summary of the issue in the Title above -->

## Environment
<!--- Provide as much information about your environment as possible: output from `k6 version`,
      OCI runtime and image information if using containers, cloud execution environment, etc. -->
- k6 version: `HEAD`
- OS and version: OSX 10.15.7

## Expected Behavior
<!--- Tell us what should happen -->
`go test -race ./...` should pass locally, on OSX

## Actual Behavior
<!--- Tell us what happens instead of the expected behavior -->
`go test -race ./...` fails for `TestDNSResolver`, steps to reproduce below. 

Some other observations:
* I think this test makes the assumption that `127.0.0.254:PORT` should return a `connect refused` if nothing is listening on that socket. This is true on a Raspbian (Debian) machine, but not on an OSX machine. My networking knowledge is fuzzy here so maybe this is just obvious. 
```
# OSX
$ curl -m 2 127.0.0.254
curl: (28) Connection timed out after 2004 milliseconds
```
```
# Raspberry PI
pi@raspberrypi:~ $ curl 127.0.0.254
curl: (7) Failed to connect to 127.0.0.254 port 80: Connection refused
```
* I _think_ 0.0.0.0 is generally considered non routable and will fail on both: https://en.wikipedia.org/wiki/0.0.0.0. I can  replace the IP we substitute into those tests if we think that's the right approach. 

## Steps to Reproduce the Problem
```
$ go test -timeout 30s -run ^TestDNSResolver$ go.k6.io/k6/core/local

--- FAIL: TestDNSResolver (23.70s)
    --- FAIL: TestDNSResolver/cache (23.58s)
        --- FAIL: TestDNSResolver/cache/0 (5.92s)
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/0
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/0
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/0
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/0
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/0
        --- FAIL: TestDNSResolver/cache/1000 (5.86s)
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/1000
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/1000
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/1000
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/1000
        --- FAIL: TestDNSResolver/cache/3s (5.83s)
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/3s
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/3s
            /Users/vishal.kuo/go/src/github.com/loadimpact/k6/core/local/local_test.go:1066: 
                	Error Trace:	local_test.go:1066
                	Error:      	Error message not equal:
                	            	expected: "dial tcp 127.0.0.254:54167: connect: connection refused"
                	            	actual  : "context deadline exceeded"
                	Test:       	TestDNSResolver/cache/3s
FAIL
FAIL	go.k6.io/k6/core/local	24.738s
FAIL
```

--- a/core/local/local_test.go
+++ b/core/local/local_test.go
@@ -1035,7 +1035,7 @@ func TestDNSResolver(t *testing.T) {
 		}
 
 		expErr := sr(`dial tcp 127.0.0.254:HTTPBIN_PORT: connect: connection refused`)
-		if runtime.GOOS == "windows" {
+		if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
 			expErr = "request timeout"
 		}
 		for name, tc := range testCases {

